### DISCUSSION TITLE
How do you delete a single Media item?

### DISCUSSION THREAD

**Discussion by mattpramschufer** [spatie/laravel-medialibrary#1908] [Q&A] [answered]
How do I delete a single media item from it's ID?  Right now I have

`Media::where('id', $request->get('document_id'))->delete();`

Which deletes the DB record, but it does NOT delete the files.

Any suggestions would be helpful, I looked all through the docs and didn't see anywhere it referenced deleting of single item.

**Comment by catalinux** [comment]
You can add a listener for the delete event of that model

**Reply by mattpramschufer** [reply]
So there is no built in way to delete a single media item and file from an ID then?

**Comment by miclf** [comment, marked as answer]
**EDIT:**
The package uses [an observer on the Eloquent model](https://github.com/spatie/laravel-medialibrary/blob/master/src/MediaCollections/Models/Observers/MediaObserver.php). This is where the deletion of files take place.

My _guess_ is that, in order for this to work, you need to call `delete()` _directly on an instance of your media model_, and not on the query builder as you do it in your example. I think you need a real Eloquent instance to call the `delete()` method on in order for the observer to be triggered.

Does this work?
```php
Media::find($request->get('document_id'))->delete();
```
---
**Original reply:**
In my controller, I call
```php
Media::find($request->input('photo'))->delete();
```
to delete a media from its ID.

Otherwise, I see that the `InteractsWithMedia` trait contains these two methods:
```php
/**
 * Delete the associated media with the given id.
 * You may also pass a media object.
 *
 * @param int|\Spatie\MediaLibrary\MediaCollections\Models\Media $mediaId
 *
 * @throws \Spatie\MediaLibrary\MediaCollections\Exceptions\MediaCannotBeDeleted
 */
public function deleteMedia($mediaId): void
{
    // ...
}

public function deletePreservingMedia(): bool
{
    // ...
}
```
